Package.swift

```diff
@@ -47,7 +49,7 @@ let package = Package(
                 "Build",
                 "Xcodeproj",
                 "Workspace"
-            ]
+            ] + (buildPackageSyntax ? ["PackageSyntax"] : [])
```

Comment:
Would it make sense to add these conditionals to the bottom of the manifest, along the lines of the other conditional dependencies etc?  That would keep the top of the manifest parseable.  That sort of goes to the heart of the question of using SwiftSyntax in the first place — is it useful to recommend a certain manifest layout to make it as parseable as possible, even when that means that additional configuration "below the fold" of the manifest is unseen?

Reply:
I don't think this is possible since `PackageDescription.Product` doesn't have a public `targets` property that could be used to append new ones after it's been initialized. The good news is that even though the proposal doesn't guarantee it, the current editor implementation is able to recognize array concatenation like this and would still be able to add new targets to the list.

> is it useful to recommend a certain manifest layout to make it as parseable as possible, even when that means that additional configuration "below the fold" of the manifest is unseen?

The current proposal test attempts to define what a "fully declarative" manifest looks like, one of the advantages being that any fully declarative manifest (+ some partially declarative ones) is guaranteed to be machine-editable. It might make sense to provide a linter to verify this at some point.

Reply:
> I don't think this is possible since `PackageDescription.Product` doesn't have a public `targets` property that could be used to append new ones after it's been initialized.

Ah, right — each of the product types for which it's relevant has a `targets` property, but it's currently immutable.  That might possibly be good to fix.

> The good news is that even though the proposal doesn't guarantee it, the current editor implementation is able to recognize array concatenation like this and would still be able to add new targets to the list.

That's great to hear!

Given that, this seems good to me.  If we can make the product type properties mutable, we can maybe make this a little cleaner in the future.  Thanks for explaining!